e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/remove-fees-from-payment.spec.js

```diff
@@ -0,0 +1,122 @@
+
+  const utilisationReport = UtilisationReportEntityMockBuilder.forStatus(UTILISATION_REPORT_RECONCILIATION_STATUS.RECONCILIATION_IN_PROGRESS)
+    .withId(reportId)
+    .withBankId('961')
+    .withDateUploaded(new Date())
+    .build();
+
+  const aPaymentWithAmount = (amount) =>
+    PaymentEntityMockBuilder.forCurrency(paymentCurrency)
+      .withId(paymentId)
+      .withAmount(amount)
+      .withDateReceived(datePaymentReceived)
+      .withReference(paymentReference)
+      .build();
+
+  const aFeeRecordWithAmountStatusAndPayments = (id, amount, status, payments) =>
+    FeeRecordEntityMockBuilder.forReport(utilisationReport)
+      .withId(id)
+      .withStatus(status)
+      .withFeesPaidToUkefForThePeriod(amount)
+      .withFeesPaidToUkefForThePeriodCurrency(paymentCurrency)
+      .withPaymentCurrency(paymentCurrency)
+      .withPayments(payments)
+      .build();
+
+  beforeEach(() => {
+    cy.task(NODE_TASKS.REMOVE_ALL_UTILISATION_REPORTS_FROM_DB);
+    cy.task(NODE_TASKS.REMOVE_ALL_PAYMENTS_FROM_DB);
+
+    cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [utilisationReport]);
+
+    const payment = aPaymentWithAmount(paymentAmount);
+
+    const feeRecords = feeRecordIds.map((id) => aFeeRecordWithAmountStatusAndPayments(id, paymentAmount, FEE_RECORD_STATUS.MATCH, [payment]));
+    cy.task(NODE_TASKS.INSERT_FEE_RECORDS_INTO_DB, feeRecords);
+
+    pages.landingPage.visit();
+    cy.login(USERS.PDC_RECONCILE);
+
+    cy.visit(`/utilisation-reports/${reportId}/edit-payment/${paymentId}`);
+  });
+
+  it('should display errors when form submitted with invalid fee selections and persist the selected fees', () => {
+    pages.utilisationReportEditPaymentPage.clickRemoveSelectedPaymentsButton();
+    pages.utilisationReportEditPaymentPage.errorSummary().contains('Select fee or fees to remove from the payment');
+
+    feeRecordIds.forEach((id) => {
+      cy.get(`[type="checkbox"][id="feeRecordId-${id}"]`).check();
+    });
+    pages.utilisationReportEditPaymentPage.clickRemoveSelectedPaymentsButton();
+    pages.utilisationReportEditPaymentPage.errorSummary().contains('You cannot remove all the fees. Delete the payment instead.');
+    feeRecordIds.forEach((id) => {
+      cy.get(`[type="checkbox"][id="feeRecordId-${id}"]`).should('be.checked');
+    });
+  });
+
+  it('should remove the selected fees from the payment after clicking the remove selected fees button', () => {
+    const feeRecordId = feeRecordIds[0];
+
+    cy.get(`[type="checkbox"][id="feeRecordId-${feeRecordId}"]`).check();
+    pages.utilisationReportEditPaymentPage.clickRemoveSelectedPaymentsButton();
+
+    cy.url().should('eq', relative(`/utilisation-reports/${reportId}/edit-payment/${paymentId}`));
+    cy.get(`[type="checkbox"][id="feeRecordId-${feeRecordId}"]`).should('not.exist');
+  });
+
+  it(`should update the fee record status if another fee has been removed from the payment group`, () => {
+    const feeRecordId = feeRecordIds[0];
```

Comment:
It looks like this is only being used in the checkbox selection below, but isn't actually being used to create either of the fee records below. I also don't think this test needs to rely on `feeRecordIds` and you can probably just hardcode `feeRecordId` to be some set value

Reply:
Yep use of feeRecordIds is unnecessary - resolved in 46ba4e10dd173f6bcd9d4388c168db454fa6191e thank you